aesd-char-driver/main.c

```diff
@@ -48,21 +58,93 @@ ssize_t aesd_read(struct file *filp, char __user *buf, size_t count,
                 loff_t *f_pos)
 {
 	ssize_t retval = 0;
+	ssize_t offset = 0;
+	ssize_t bytes_to_user = 0;
+	ssize_t count_bytes = 0;
+	struct aesd_buffer_entry * read_entry = NULL;
+	struct aesd_dev *dev = filp->private_data;
 	PDEBUG("read %zu bytes with offset %lld",count,*f_pos);
 	/**
 	 * TODO: handle read
 	 */
+	
+	if (mutex_lock_interruptible(&dev->lock))
```

Comment:
@Jahnavi-pinnamaneni do you think this is proper error check for mutex lock?

Reply:
@swatikadivar, I made use of mutex_lock_interruptible since, I wanted to avoid going into indefinite wait state.